README.md

```diff
@@ -1 +1,250 @@
+# nqm-tdx-terminal-cli
+Nquiringminds TDX API client application
+
+## Install
+The best use of tdxcli tool is to install it with npm globally as follows:
+```bash
+npm i -g nqm-tdx-terminal-cli
+```
+The client app can be accessed by running the command ```tdxcli```.
+
+## Usage
+```bash
+Usage: tdxcli <command> [options]
+
+Commands:
+  tdxcli signin [id] [secret]                   Sign in to tdx
+  tdxcli signout                                Sign out of tdx
+  tdxcli info [type] [id]                       Output current account info
+  tdxcli config                                 Output tdx config
+  tdxcli list                                   List all configured aliases
+  tdxcli runapi <command>                       Run a tdx api command
+  tdxcli download <id> [filepath]               Download resource
+  tdxcli upload <id> <filepath>                 Upload resource
+  tdxcli copyalias <aliasname>                  Makes a copy of an existing alias configuration
+  tdxcli modifyalias <aliasname> <configjson>   Modifies an existing alias configuration
+  tdxcli removealias <aliasname>                Removes an existing alias configuration
+  tdxcli abortdatabot <instanceid>              Aborts a databot instance
+  tdxcli stopdatabot <instanceid>               Stops a databot instance
+  tdxcli startdatabot <databotid> <configjson>  Starts a databot instance
+
+Options:
+  -a, --alias    Alias name                                                                   [string]
+  -h, --help     Show help                                                                   [boolean]
+  -v, --version  Show version number                                                         [boolean]
+```
+
+## Documentation
+In order to use the ```tdxcli``` app one has to sign into a tdx account with an email address or share token (id and secret). If the user signs in using an email address the ```tdxcli``` will automatically open a Chromium browser window where the user can input the credentials. If the user signs in with an email id + secret the ```tdxcli``` app will open a headless Chromium window and will automatically fill in the credentials. Finally, if the user signs in with a share token the ```tdxcli``` will sign in using the tdx api authentication method.
+
+### ```signin```
+Usage
+```bash
+tdxcli signin
+tdxcli signin emailorsharetokenid thesecret
+```
+The first command will open a Chromium browser window, wheares the second will do an automatic signin with the provided credentials. The obtained access token will be stored in the ```.env``` file.
+
+Initially the user has to choose an ```alias``` in order to sign into a given tdx account. The default aliases are ```nqminds``` and ```nq_m```, which correspond to ```tdx.nqminds.com``` and ```tdx.nq_m.com```, respectively .
+
+```bash
+tdxcli signin --alias=nqminds
+tdxcli signin emailorsharetokenid thesecret --alias=nq_m
+```
+
+The aliases configurations are stored in ```config.json```:
```

Comment:
Instead of using `config.json`, it might be better to follow: https://github.com/yargs/yargs/blob/master/docs/advanced.md#building-configurable-cli-apps

Yargs can then read `.rc` files, or options in `package.json`, just like eslint.